packages/core/test/acceptance/di_spec.ts

```diff
@@ -3546,4 +3546,430 @@ describe('di', () => {
+        }
+      }
+
+      @Component({
+        template: `
+          <ng-template>
+            <menu></menu>
+          </ng-template>
+        `
+      })
+      class App {
+        @ViewChild(TemplateRef) template!: TemplateRef<unknown>;
+      }
+
+      @NgModule({
+        declarations: [App, Menu],
+        exports: [App, Menu],
+        providers: [{provide: token, useValue: 'hello from module'}]
+      })
+      class Module {
+      }
+
+      TestBed.configureTestingModule({imports: [Module]});
+      const injector =
+          Injector.create({providers: [{provide: token, useValue: 'hello from injector'}]});
+      const fixture = TestBed.createComponent(App);
+      fixture.detectChanges();
+
+      fixture.componentInstance.template.createEmbeddedView({}, injector);
+      fixture.detectChanges();
+
+      expect(injectedValue).toBe('hello from injector');
+    });
+
+    it('should use a custom injector when the view is created outside of the declaration view',
+       () => {
+         const declarerToken = new InjectionToken<string>('declarerToken');
+         const creatorToken = new InjectionToken<string>('creatorToken');
+
+         @Directive({selector: 'menu'})
+         class Menu {
+           constructor(
+               @Inject(token) public tokenValue: string,
+               @Optional() @Inject(declarerToken) public declarerTokenValue: string,
+               @Optional() @Inject(creatorToken) public creatorTokenValue: string) {}
+         }
+
+         @Component({
+           selector: 'declarer',
+           template: '<ng-template><menu></menu></ng-template>',
+           providers: [{provide: declarerToken, useValue: 'hello from declarer'}]
+         })
+         class Declarer {
+           @ViewChild(Menu) menu!: Menu;
+           @ViewChild(TemplateRef) template!: TemplateRef<unknown>;
+         }
+
+         @Component({
+           selector: 'creator',
+           template: '<menu-trigger></menu-trigger>',
+           providers: [{provide: creatorToken, useValue: 'hello from creator'}]
+         })
+         class Creator {
+           @ViewChild(MenuTrigger) trigger!: MenuTrigger;
+         }
+
+         @Component({
+           template: `
+              <declarer></declarer>
+              <creator></creator>
+            `
+         })
+         class App {
+           @ViewChild(Declarer) declarer!: Declarer;
+           @ViewChild(Creator) creator!: Creator;
+         }
+
+         TestBed.configureTestingModule(
+             {declarations: [App, MenuTrigger, Menu, Declarer, Creator]});
+         const injector = Injector.create({providers: [{provide: token, useValue: 'hello'}]});
+         const fixture = TestBed.createComponent(App);
+         fixture.detectChanges();
+
+         fixture.componentInstance.creator.trigger.menu =
+             fixture.componentInstance.declarer.template;
+         fixture.componentInstance.creator.trigger.open(injector);
+         fixture.detectChanges();
```

Comment:
nit: I think it'd be a bit easier to read if we create a new vars for some values:
```suggestion
         const {creator, declarer} = fixture.componentInstance;
         creator.trigger.menu = declarer.template;
         creator.trigger.open(injector);
         fixture.detectChanges();
```